examples/text-input.c

```diff
@@ -0,0 +1,197 @@
+#include <GLES2/gl2.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <wayland-client.h>
+#include <wayland-egl.h>
+#include <wlr/render/egl.h>
+#include "text-input-unstable-v3-client-protocol.h"
+#include "xdg-shell-client-protocol.h"
+
+#include <linux/input-event-codes.h>
```

Comment:
Add in FreeBSD too. You can find examples of how to do this by grepping other examples.